lib/utils/addEntries.js

```diff
@@ -120,16 +120,37 @@ function addEntries(compiler, options, server) {
   // eslint-disable-next-line no-shadow
   compilers.forEach((compiler) => {
     const config = compiler.options;
+    const target = config.target;
     /** @type {Boolean} */
-    const webTarget = [
-      'web',
-      'webworker',
-      'electron-renderer',
-      'node-webkit',
-      // eslint-disable-next-line no-undefined
-      undefined,
-      null,
-    ].includes(config.target);
+    let webTarget = false;
+    try {
+      // only available in webpack@5
+      const {
+        getTargetsProperties,
+        getTargetProperties,
+        // @ts-ignore
+        // eslint-disable-next-line import/no-unresolved
+      } = require('webpack/lib/config/target');
+      if (target) {
+        const targetProperties =
+          typeof target === 'string'
+            ? getTargetProperties(target, config.context)
+            : getTargetsProperties(target, config.context);
```

Comment:
It'd be cleaner if webpack updates `options` with `targetProperties`, but I cannot seem to find it.

Reply:
It seems there is an existing PR: https://github.com/webpack/webpack/pull/11738.

/cc @chenxsan

Reply:
I can continue it if I get the green light.

Reply:
I request @sokra to review it and do release ASAP